src/pages/api/user/updateUser.ts

```diff
@@ -0,0 +1,43 @@
+import { PrismaClient } from '@prisma/client';
+
+export default async function handler(req, res) {
+    const prisma = new PrismaClient();
+
+    // Check the HTTP method and if we have a body
+    if (req.method !== 'POST') {
+        return res.status(405).json({message: 'HTTP method not valid - only POST Accepted'});
+      }
+    if (!req.body) {
+    return res.status(404).json({error: "Don't have form data."});
+    }
+
+    // Get the data from the body (just name and type for now)
+    const {email, name, type} = req.body;
+
+    if (!type || type === '' || !email) {
+        return res.status(422).json({message: 'User type or email is missing'});
+    }
+    let data;
+    if (name) {
+        data = {
+            name: name,
+            type: type,
+        }
+    }
+    else {
+        data = {
+            type: type,
+        }
+    }
+    const updateUser = await prisma.user.update({
+        where: {
+            email: email,
```

Comment:
Should we validate with userId instead of email? I've been torn, because they are both unique fields. And I didn't see a way to change email in the form you made, but that is something we'd want to consider

Reply:
If you do, you can get token info with getToken(). I think it's cheaper than getServerSession()

Reply:
I just already had email on hand so I decided to use it. But definitely userId is the more robust approach